src/Sylius/Bundle/CoreBundle/Doctrine/Migrations/AbstractMySqlMigration.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ *  This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Doctrine\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+abstract class AbstractMySqlMigration extends AbstractMigration
+{
+    public function preUp(Schema $schema): void
+    {
+        $this->abortIf(!$this->isMySql(), 'Migration can only be executed safely on \'mysql\'.');
+    }
+
+    public function preDown(Schema $schema): void
+    {
+        $this->abortIf(!$this->isMySql(), 'Migration can only be executed safely on \'mysql\'.');
+    }
+
+    protected function isMariaDb(): bool
```

Comment:
I have some doubts about the naming of this class - we have `AbstractMySqlMigration`, which has methods `isMySql` and `isMySql` - we did something wrong with the naming here

Reply:
My first thought was
> (with a little bit simplification) we can tell MariaDB is a MySQL, but MySQL is not a MariaDB.

Due to that fact I put these methods in a single class. Now when you noticed it I think we can just call it `AbstractMigration` and extend the base one.